src/spectral.jl

```diff
@@ -167,3 +129,154 @@ function incidence_matrix(g::SimpleGraph, T::DataType=Int)
     spmx = SparseMatrixCSC(n_v,n_e,colpt,rowval,nzval)
     return spmx
 end
+
+"""
+Given two oriented edges i->j and k->l in g, the
+non-backtraking matrix B is defined as
+
+B[i->j, k->l] = δ(j,k)* (1 - δ(i,l))
+
+returns a matrix B, and an edgemap storing the oriented edges' positions in B
+"""
+function non_backtracking_matrix(g::SimpleGraph)
+    # idedgemap = Dict{Int, Edge}()
+    edgeidmap = Dict{Edge, Int}()
+    m = 0
+    for e in edges(g)
+        m += 1
+        edgeidmap[e] = m
+    end
```

Comment:
This is the code that @CarloLucibello wrote.
The next section uses `m` after the loop in order to add in the reverse edges?